### PR TITLE
DBT-1305 Supabase

### PR DESCRIPTION
**Pull request by Mrtn-Omwenga**
This branch focused on testing the auth-supabase service against the staging environment set up in supabase. 

The changes suggested in [this PR](https://github.com/vocali-genesis/admin-portal/pull/1) have been implemented as far as the contents of this branch allow. I have also implemented a static error handler that just logs the error for now. In the `register-login-page` branch, I will implement react-hot-toast.

All test cases have been added to `.gitignore` due to concerns raised #1. Unit tests for the service layer that use mock will be written later.

### CODE REVIEW

File: core/error-handler.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// import { toast } from 'react-toastify';
+
+export default (error: string) => {
+  console.error("Error occurred:", error);
+  // toast.error(error.message || "An unexpected error occurred");
```
Comment (abelbordonado) [inline core/error-handler.ts:5]: Sentry will go here I guess!

File: services/auth/auth-supabase.service.ts [hunk 2]
```diff
@@ -10,23 +11,34 @@ class AuthService {
     this.supabase = createClient(supabaseUrl, supabaseAnonKey);
   }
 
-  async registerUser(email: string, password: string): Promise<{ user: User | null; token: string | undefined }> {
+  private handleError(error: any): { user: null; token: undefined } {
```
Comment (abelbordonado) [inline services/auth/auth-supabase.service.ts:14]: Lets avoid `any` at any cost.

Here we can go by `Error`  (the (catch(error) type is always unknown so always requires to down cast)

File: services/auth/auth-supabase.service.ts [hunk 3]
```diff
@@ -10,23 +11,34 @@ class AuthService {
...
+    errorHandler(error.message);
+    return { user: null, token: undefined };
```
Comment (abelbordonado) [inline services/auth/auth-supabase.service.ts:16]: The handle error shall not return anything. but the function it self. Different functions may require different returns

Actually we may not need this function but can call`errorHandler` directly

File: core/error-handler.ts (hunk 1 above)
Comment (Mrtn-Tianlu) [reply core/error-handler.ts:5]: I've implemented react hot toast in the register-login-page branch. We can change though.

**Review by abelbordonado — APPROVED** [review]
Approved, lets move ahead!

**Pull request merged**